onbuild/r2d_overlay.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python3
+import subprocess
+import sys
+import os
+import argparse
+import re
+
+NB_UID = int(os.environ.get('NB_UID', 1000))
+REPO_DIR = os.environ['REPO_DIR']
+# We copy contents of *child* image to a subdirectory temporarily
+# This helps us apply customizations that were only from the
+# child, and re-do the packages of the parent.
+ONBUILD_CONTENTS_DIR = os.path.join(REPO_DIR, '.onbuild-child')
+
+
+def become(uid):
+    def wrap(func):
+        def _pre_exec():
+            # TODO: Look at the security of this very, very carefully
+            os.setgid(uid)
+            os.setuid(uid)
+        func._pre_exec = _pre_exec
+        return func
+    return wrap
+
+
+def binder_path(path):
+    if os.path.exists(os.path.join(ONBUILD_CONTENTS_DIR, 'binder')):
+        return os.path.join(ONBUILD_CONTENTS_DIR, 'binder', path)
+    return os.path.join(ONBUILD_CONTENTS_DIR, path)
+
+
+@become(NB_UID)
+def apply_environment():
+    env_path = binder_path('environment.yml')
+    if os.path.exists(env_path):
+        return [
+            f'conda env update -n root -f {env_path}',
+            f'conda clean -tipsy',
+            f'conda list -n root',
+            f'rm -rf /srv/conda/pkgs'
+        ]
+
+
+@become(NB_UID)
+def apply_requirements():
+    req_path = binder_path('requirements.txt')
+    env_path = binder_path('environment.yml')
+
+    if os.path.exists(req_path) and not os.path.exists(env_path):
```

Comment:
what is the intended behavior when both requirements.txt and environment.yml are present?

Reply:
requirements.txt is ignored. That's what repo2docker does (I think)